src/core/execute.c

```diff
@@ -168,13 +168,19 @@ void exec_context_tty_reset(const ExecContext *context, const ExecParameters *p)
         else if (lock_fd < 0)
                 log_warning_errno(lock_fd, "Failed to lock /dev/console, proceeding without lock: %m");
 
-        if (context->tty_vhangup)
-                (void) terminal_vhangup_fd(fd);
-
         if (context->tty_reset)
                 (void) terminal_reset_defensive(fd, /* switch_to_text= */ true);
 
-        (void) exec_context_apply_tty_size(context, fd, fd, path);
+        r = exec_context_apply_tty_size(context, fd, fd, path);
+        if (r < 0)
+                log_debug_errno(r, "Failed to configure TTY dimensions, ignoring: %m");
+
+        if (context->tty_vhangup)
+                (void) terminal_vhangup_fd(fd);
```

Comment:
I'm pretty sure vhangup() resets the size. I fixed a similar bug in util-linux a while ago

Reply:
yeah, but that's fine really. the hangup is optional here after all, and this reset function is mostly about "destructive" reset, i.e. to clear any leftovers from a previous session. 

It is not the "constructive" reset, which are our preparations for what comes next. We'll initialize the tty size there again, and we keep the fd open then and issue no hangups. 

I added longer comments about constructive/destructive reset in another commit of this PR, to explain this a bit.